x-pack/legacy/plugins/siem/server/lib/detection_engine/rules/add_rule_id_to_tags.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { INTERNAL_RULE_ID_KEY } from '../../../../common/constants';
+
+export const addRuleIdToTags = (tags: string[], ruleId: string | null | undefined): string[] => {
+  if (ruleId == null) {
+    return tags;
+  } else {
+    return [...tags, `${INTERNAL_RULE_ID_KEY}:${ruleId}`];
```

Comment:
It might be (slightly) more performant here to utilize `concat` to append the internal rule id tag.

Reply:
Per conversations we will keep using the ES2015+ style unless we are dealing with larger arrays.

Thanks for pointing out though that technically the concat is faster as I typically like that style more but the ES2015 is what we encourage on the team for small stuff.